rskj-core/src/test/java/co/rsk/peg/FederationTestUtils.java

```diff
@@ -269,10 +266,29 @@ public static void spendFromP2shP2wshErpStandardFed(
         TransactionWitness txWitness = TransactionWitness.createWitnessErpScript(redeemScript, txSignatures);
         spendTx.setWitness(0, txWitness);
 
+        String rawTx = Hex.toHexString(spendTx.bitcoinSerialize());
+        double estimatedSize = calculateTxSize(txWitness, rawTx);
+
+        Coin fee = prevValue.minus(value);
+        Coin estimatedFeeRate = fee.divide((long) estimatedSize);
+
         // Uncomment to print the raw tx in console and broadcast https://blockstream.info/testnet/tx/push
         System.out.println(Hex.toHexString(spendTx.bitcoinSerialize()));
     }
 
+    public static double calculateTxSize(TransactionWitness txWitness, String rawTx) {
+        int estimatedTotalSize = rawTx.length()/2;
+        int estimatedWitnessSize = 0;
+        for (int i = 0; i < txWitness.getPushCount(); i++) {
+            estimatedWitnessSize += txWitness.getPush(i).length;
+        }
+        int estimatedBaseSize = (estimatedTotalSize - estimatedWitnessSize);
+        double estimatedTxWeight = estimatedTotalSize + 3*estimatedBaseSize;
+        double estimatedTxVirtualSize = Math.max(0.25*estimatedTotalSize + 0.75*estimatedBaseSize, estimatedTxWeight/4);
+
+        return estimatedTxVirtualSize;
+    }
```

Comment:
```suggestion
    public static double calculateTxSize(TransactionWitness txWitness, String rawTx) {
        int estimatedTotalSize = rawTx.length() / 2;
        int estimatedWitnessSize = 0;
        for (int i = 0; i < txWitness.getPushCount(); i++) {
            estimatedWitnessSize += txWitness.getPush(i).length;
        }
        int estimatedBaseSize = estimatedTotalSize - estimatedWitnessSize;
        double estimatedTxWeight = (estimatedTotalSize + 3) * estimatedBaseSize; // comment to explain why 3 is added to estimatedTotalSize
        double estimatedTxVirtualSize = Math.max(0.25 * estimatedTotalSize + 0.75 * estimatedBaseSize, estimatedTxWeight / 4); // comment to explain 0.25, 0.75

        return estimatedTxVirtualSize;
    }
```

Reply:
the estimatedTxWeight is not `(estimatedTotalSize + 3) * estimatedBaseSize`,
it is `estimatedTotalSize + (3 * estimatedBaseSize)`
the other comments are accurate, i will add the suggested commented code after some tests that support that fee calculation.
thanks kelvin!

Reply:
Well Done